### PR TITLE
Fix `connect` example for UDP

### LINKED ISSUE #240: `connect` example panics on udp
The example panics when trying to connect to a udp port (`cargo run --example connect -- 127.0.0.1:8080 --udp`) with the following message:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: SpawnError { is_shutdown: true }', libcore/result.rs:945:5
```

It happens because [`tokio::spawn`](https://github.com/tokio-rs/tokio/blob/master/examples/connect.rs#L201) is called before the runtime has been initialized (with [`tokio::run`](https://github.com/tokio-rs/tokio/blob/master/examples/connect.rs#L73)).

(both links lead to the example's code)

### PR DESCRIPTION
**Pull request by mexus**
I've postpone the *spawn* of the stream that forwards data from *stdin* to the connection by utilizing `future::lazy`. It's the most easy way to fix the example (#240) I was able to come up with.

Closes  #240.

### CODE REVIEW

**Comment by mexus** [comment]
Travis fails on stable on the `hammer_split` test:
`test hammer_split has been running for over 60 seconds`

I've tested it locally and it seems like there is an issue in that test since sometimes (rather rarely though) the aforementioned test fails with `signal: 11, SIGSEGV: invalid memory reference`.

File: examples/connect.rs [hunk 1]
```diff
@@ -198,24 +198,29 @@ mod udp {
...
             if src == addr {
                 Some(chunk.into())
             } else {
                 None
             }
-        }))
+        });
+
+        Box::new(future::lazy(|| {
+            tokio::spawn(forward_stdin);
```
Comment (kpp) [inline examples/connect.rs]: Why not `tokio::spawn( forward_stdin.and_then(|_| receive) )`?

Comment (kpp) [reply examples/connect.rs]: You'd better call `tokio::spawn` or `tokio::run` in main, here it would be better to return a pure future only.

Comment (kpp) [reply examples/connect.rs]: select/join

**Comment by carllerche** [comment]
@mexus Re the hammer test failing, this is a known issue and I am currently working on it :)

File: examples/connect.rs (hunk 1 above)
Comment (kpp) [reply examples/connect.rs]: Ah, sorry. You are right.

**Comment by mexus** [comment]
@carllerche thanks :) @kpp has already explained the situation to me briefly, so I've managed to make Travis happy :-)

File: examples/connect.rs (hunk 1 above)
Comment (mexus) [reply examples/connect.rs]: Ok, cool :)

**Pull request merged**